localstack-core/localstack/utils/id_generator.py

```diff
@@ -16,6 +17,12 @@ def set_custom_id_by_unique_identifier(self, unique_identifier: str, custom_id:
         with self._lock:
             self._custom_ids[unique_identifier] = custom_id
 
+    @contextmanager
+    def custom_id(self, resource_identifier: ResourceIdentifier, custom_id: str) -> None:
+        yield self.set_custom_id(resource_identifier, custom_id)
+
+        self.unset_custom_id(resource_identifier)
```

Comment:
We probably want to wrap this in a try/finally block to make sure that we reset the state if an error is thrown
Can we also cover this in the tests?

Reply:
Good thinking! I will add that